src/renderer/views/library/commands.vue

```diff
@@ -3,20 +3,126 @@
+            <Select v-model="sortMethod" :options="sortMethods" label="Sort By" />
+        </div>
+
+        <div class="flex-col flex-grow fullheight">
+            <div class="scroll-container">
+                <div v-for="command in filteredCommands" :key="command.cmd" class="command">
+                    <div class="cmd">{{ command.cmd }}</div>
+                    <div class="cmdDescription">{{ command.cmdDescription }}</div>
+                </div>
+            </div>
+        </div>
     </div>
 </template>
 
 <script lang="ts" setup>
-import Markdown from "vue3-markdown-it";
+import { computed, onMounted, onUnmounted, reactive, Ref, ref } from "vue";
+
+import SearchBox from "@/components/controls/SearchBox.vue";
+import Select from "@/components/controls/Select.vue";
+
+type SortMethod = "Name" | "Description";
+
+const sortMethods: SortMethod[] = ["Name", "Description"];
+const sortMethod: Ref<SortMethod> = ref("Name");
+
+interface Command {
+    cmd: string;
+    cmdDescription: string;
+}
+
+const searchVal = ref("");
+
+const commands = reactive<Command[]>([]);
+
+// Sort the commands array based on the sort method
+function sortCommands(commands: Command[], sortMethod: SortMethod) {
+    switch (sortMethod) {
+        case "Name":
+            return commands.sort((a, b) => a.cmd.localeCompare(b.cmd));
+        case "Description":
+            return commands.sort((a, b) => a.cmdDescription.localeCompare(b.cmdDescription));
+        default:
+            return commands;
+    }
+}
+
+const filteredCommands = computed(() => {
+    if (!searchVal.value) {
+        return sortCommands(commands, sortMethod.value);
+    }
+    return sortCommands(
+        commands.filter((command) => (command.cmd + command.cmdDescription).includes(searchVal.value)),
+        sortMethod.value
+    );
+});
 
 const route = api.router.currentRoute.value;
+
+onMounted(() => {
+    // Listen for the response from the server
+    api.comms.onResponse("s.communication.received_direct_message").add(async (data) => {
+        const { message } = data;
+        // Check if the message is a command
+        if (!message.startsWith("!") && !message.startsWith("$")) return;
+
+        const cmd = message.split("-")[0].split(" ")[0];
+        const cmdDescription = message.slice(cmd.length + 1).replace("-", " ");
+
+        cmdDescription && !cmdDescription.includes("*") && commands.push({ cmd, cmdDescription });
+    });
+    // Send a message to the server to get all the commands
+    api.comms.request("c.communication.send_direct_message", {
+        recipient_id: 3137,
+        message: `!helpall`,
+    });
+    api.comms.request("c.communication.send_direct_message", {
+        recipient_id: 3137,
+        message: `$help`,
+    });
+});
+
+// Unsubscribe from the response listener when the component is not visible anymore
+onUnmounted(() => {
+    /**
+  I don't know how to unsubscribe from the response listener
```

Comment:
`.onResponse` returns a `SignalBinding` instance that you can later call `.destroy()` on to solve this. See `HostBattle.vue` for an example